frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/task-sidebar/TaskSidebarResource.tsx

```diff
@@ -12,28 +12,49 @@ type TaskSidebarResourceProps = {
   resource: TektonResource;
 };
 
+interface ResourceLink {
+  name: string;
+  resource: string;
+}
+
 const TaskSidebarResource: React.FC<TaskSidebarResourceProps> = (props) => {
   const { t } = useTranslation();
-  const { setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
+  const { getFieldMeta, setFieldValue } = useFormikContext<PipelineBuilderFormikValues>();
   const {
     availableResources,
     hasResource,
     name,
-    resource: { name: resourceName, type: resourceType },
+    resource: { name: resourceName, type: resourceType, optional = false },
   } = props;
 
   const dropdownResources = availableResources.filter(
     (resource) => resourceType === resource.type && !!resource.name?.trim(),
   );
+
+  const currentLinkedResourceName = getFieldMeta<ResourceLink>(name).value?.resource;
+  const currentLinkedResourceSelectable =
+    currentLinkedResourceName &&
+    dropdownResources.some((resource) => resource.name === currentLinkedResourceName);
+
   const options: FormSelectFieldOption[] = [
     {
       label: t('pipelines-plugin~Select {{resourceType}} resource...', { resourceType }),
       value: '',
       isPlaceholder: true,
-      isDisabled: true,
```

Comment:
Disabled when required, not disabled when optional? I think that's better than just enabling the placeholder option.

Maybe we also change the label to indicate optional 🤔   Maybe something like `No {{resourceType}} resource` to mimic what we do in the Start/Trigger Modals.